lib/default.nix

```diff
@@ -29,7 +32,7 @@ let
     versions = callLibs ./versions.nix;
 
     # module system
-    modules = callLibs ./modules.nix;
+    modules = import ./modules.nix  { lib = self; inherit finalLib; }; # module argument needs the final lib
```

Comment:
Is `self` not equal to `finalLib`? If equal, `finalLib` can be omitted here. If not, either the naming is not according to convention, or there's a bug somewhere.